### PR TITLE
Webhook simple cert

### PR DESCRIPTION
**Pull request by arturrez**

### CODE REVIEW

File: controllers/common/webhook.go [hunk 1]
```diff
@@ -67,36 +70,151 @@ func (w Webhook) SetupWebhookDeps() error {
...
-	go func() {
-		ticker := time.NewTicker(resyncFreq)
-		defer ticker.Stop()
-		ctxDone := ctx.Done()
-		for {
-			l.Info("Background cert operation is starting. Scheduled every " + fmt.Sprint(resyncFreq))
-			if err := w.GenerateAndSignCertForService(ctx,
+	//use caBundle for EKS
+	useCaBundle, _ := commonEnsure.isEKS(l)
+	if useCaBundle {
```
Comment (adutchak) [inline controllers/common/webhook.go:82]: Add error handling

Comment (arturrez) [reply controllers/common/webhook.go:82]: added

File: controllers/common/webhook.go [hunk 2]
```diff
@@ -117,182 +99,84 @@ func (w Webhook) GenerateAndSignCertForService(
...
-		log.Info("Certificate signing request", "Certificate is about to expire", expired)
-
-		//check if cert is signed by k8s CA
-		k8sCACert, err := utils.GetK8sCACert()
-		if err != nil {
-			w.fatal(log, err, "Certificate signing request")
-		}
-		isSigned, err := utils.IsCertSignedByCA(cert, k8sCACert)
+	for i := 0; i < len(wrc.Webhooks); i++ {
+		//wrc.Webhooks[i].ClientConfig.CABundle = []byte(base64.StdEncoding.EncodeToString([]byte(clientCertificatePEM.Bytes())))
```
Comment (yevhenvolchenko) [inline controllers/common/webhook.go:148]: remove commented code

File: controllers/common/webhook.go [hunk 3]
```diff
@@ -117,182 +99,84 @@ func (w Webhook) GenerateAndSignCertForService(
...
+		wrc.Webhooks[i].ClientConfig.CABundle = clientCertificatePEM.Bytes()
```
Comment (yevhenvolchenko) [inline controllers/common/webhook.go:149]: it should be a CA cert, not client

File: controllers/common/webhook.go [hunk 4]
```diff
@@ -30,85 +29,68 @@ type Webhook struct {
...
-				l.Error(err, "Background cert operation")
-			}
-			select {
-			case <-ticker.C:
-				continue
-			case <-ctxDone:
-				l.Info("Background cert operation is done")
-				return
-			}
+		defer wg.Done()
```
Comment (yevhenvolchenko) [inline controllers/common/webhook.go:67]: this should be a synchronous code

**Review by yevhenvolchenko — APPROVED** [review]

File: controllers/common/webhook.go [hunk 5]
```diff
@@ -67,36 +70,151 @@ func (w Webhook) SetupWebhookDeps() error {
...
 	l := log.FromContext(ctx)
-	go func() {
-		ticker := time.NewTicker(resyncFreq)
-		defer ticker.Stop()
-		ctxDone := ctx.Done()
-		for {
-			l.Info("Background cert operation is starting. Scheduled every " + fmt.Sprint(resyncFreq))
-			if err := w.GenerateAndSignCertForService(ctx,
+	//use caBundle for EKS
+	useCaBundle, _ := commonEnsure.isEKS(l)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply controllers/common/webhook.go]: yeah, it's already cleaned up

File: controllers/common/webhook.go (hunk 3 above)
Comment (arturrez) [reply controllers/common/webhook.go:149]: we don't need CA machinery if we only need one cert

File: api/v1alpha1/avalanchego_webhook.go [hunk 6]
```diff
@@ -30,6 +30,7 @@ func (r *Avalanchego) SetupWebhookWithManager(mgr ctrl.Manager) error {
 // TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
 //+kubebuilder:webhook:path=/validate-chain-avax-network-v1alpha1-avalanchego,mutating=false,failurePolicy=fail,sideEffects=None,groups=chain.avax.network,resources=avalanchegoes,verbs=create;update,versions=v1alpha1,name=vavalanchego.kb.io,admissionReviewVersions=v1
 
+//+kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingwebhookconfigurations,verbs=get;list;watch;create;update;patch
```
[the remark this answers is not in the document]
Comment (yevhenvolchenko) [reply api/v1alpha1/avalanchego_webhook.go:33]: extend this config to validator

File: controllers/common/webhook.go [hunk 7]
```diff
@@ -67,36 +70,151 @@ func (w Webhook) SetupWebhookDeps() error {
...
+					l.Info("Background cert operation is done")
+					return
+				}
 			}
+		}()
+	}
+	return nil
+}
+
+func (w Webhook) GenerateSelfSignedCertForService(
```
[the remark this answers is not in the document]
Comment (arturrez) [reply controllers/common/webhook.go:84]: done

File: controllers/common/webhook.go (hunk 4 above)
Comment (arturrez) [reply controllers/common/webhook.go:67]: fixed

**Review by yevhenvolchenko — APPROVED** [review]
LGTM

**Pull request merged**